src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/ComponentLocator.java

```diff
@@ -32,6 +33,7 @@
    * Scans through all components and locates the first one which class name matches a substring.
    * @param componentClassSubstring A case-sensitive substring of the component's desired class.
    */
+  @RequiresEdt
   public static @Nullable Component getComponentByClass(@NotNull String componentClassSubstring) {
     Predicate<Component> predicate =
         (c) -> c.getClass().toString().contains(componentClassSubstring);
```

Comment:
Nit: Sonar complains about the `(c)` parenthesis here.

Reply:
This will be addressed in a future ticket where this class gets extended